lib/cdo/user_helpers.rb

```diff
@@ -20,7 +20,7 @@ def self.generate_username(queryable, name)
     return prefix if queryable.where(username: prefix).limit(1).empty?
 
     # Throw darts to find an appropriate suffix, using it if we hit bullseye.
-    (0..1).each do |exponent|
+    (0..2).each do |exponent|
       min_index = 10**exponent
       max_index = 10**(exponent + 1) - 1
       3.times do |_i|
```

Comment:
It does increase the number of darts thrown, from throwing three darts in the intervals 0-9 and 10-99 to three darts in the intervals 0-9, 10-99, and 100-999.

There is a downside to throwing these darts, each dart is a (indexed) DB query.